WcaOnRails/app/assets/javascripts/registrations.js

```diff
@@ -36,3 +36,21 @@ onPage('registrations#index', function() {
   // This makes the interface resemble the order.
   $('.name .sortable').addClass('asc');
 });
+
+function comparePaymentDate(a, b) {
+  var elemA = $(a);
+  var elemB = $(b);
+  var paidA = elemA.data("paidDate");
+  var paidB = elemB.data("paidDate");
+  if (paidA !== "" && paidB !== "") {
+    // Both have paid, compare their last payment dates
+    return paidA.localeCompare(paidB);
+  } else if (paidA === "" && paidB === "") {
+    // None have paid, compare their registration dates
+    return elemA.data("registeredAt").localeCompare(elemB.data("registeredAt"));
```

Comment:
Hmm... does this violate transitivity? For example, consider 3 registrations:

- Jeremy registered on Monday, and has not paid yet
- Philippe registered on Wednesday, and paid on Wednesday
- Lars registered on Tuesday and paid on Thursday

This gives: Jeremy < Philippe < Lars
But it also gives: Lars < Jeremy (because Lars has paid and Jeremy has not)

I'm pretty sure whatever sorting method our table uses is not going to work in general with this.

Maybe if a competition has payment, only show the payment date column, and if a competition does not have payment, only show the registration date column? Basically, we'd be showing a "registration completion date" column, which is payment date if the comp has payment, and registration creation date if the comp does not have payment.